src/audio/copier/copier.c

```diff
@@ -367,12 +397,24 @@ static int create_dai(struct comp_dev *parent_dev, struct copier_data *cd,
 		 * }
 		 */
 		 /* get gtw node id in config data */
-		if (copier->gtw_cfg.config_length) {
-			alh_blob = (struct sof_alh_configuration_blob *)copier->gtw_cfg.config_data;
-			dai_count = alh_blob->alh_cfg.count;
-			for (i = 0; i < dai_count; i++)
-				dai_index[i] =
+		if (is_multi_gateway(node_id)) {
+			if (copier->gtw_cfg.config_length) {
```

Comment:
Is this this intentional? Before we went to this branch if config_length was non-zero, not we go only if "is_multi_gateway()" is true.

Reply:
Yes, that if is_multi_gateway(...) was added intentionally. Quote from the spec:

_... Configuration BLOB depends on the gateway type:
Single gateway: unspecified,
Multi gateway: set to AlhMultiGtwCfg,
External VAD gateway: set to ExtVadGtwCfg. ..._

So checking for blob length is not enough, it might be ExtVadGtwCfg instead of AlhMultiGtwCfg, or something unspecified for single gateway.